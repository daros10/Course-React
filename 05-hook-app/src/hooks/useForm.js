import { useState } from 'react';

export const useForm = (initialState = {}) => {
  const [formValues, setFormValues] = useState(initialState);

  const handleInputChange = (event) => {
    setFormValues({
      ...formValues,
      [event.target.name]: event.target.value,
    });
  };

  return [formValues, handleInputChange];
};
