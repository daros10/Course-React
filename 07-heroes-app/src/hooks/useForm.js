import { useState } from 'react';

export const useForm = (initialState = {}) => {
  const [formValues, setFormValues] = useState(initialState);

  const reset = () => {
    setFormValues(initialState);
  };

  const handleInputChange = (event) => {
    setFormValues({
      ...formValues,
      [event.target.name]: event.target.value,
    });
  };

  return [formValues, handleInputChange, reset];
};
