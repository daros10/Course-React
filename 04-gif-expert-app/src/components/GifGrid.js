import React from 'react';
import PropTypes from 'prop-types';
import GifGridItem from './GifGridItem';

import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);
  return (
    <>
      <h3 className='animate__animated animate__fadeIn'> {category} </h3>
      <div className='card-grid'>
        {loading && (
          <p className='animate__animated animate__flash'>Loading...</p>
        )}
        <ol>
          {images.map((img) => (
            <GifGridItem key={img.id} img={img} />
          ))}
        </ol>
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
