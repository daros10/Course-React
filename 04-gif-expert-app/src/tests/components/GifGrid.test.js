import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import GifGrid from '../../components/GifGrid';

import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en componente GifGrid', () => {
  const category = 'Rick and Morty';

  test('Deberia mostar el componente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar items cuando se hace la llamada con el useFetchGifs', () => {
    // creo un mock del useFetchGifs

    const gifs = [
      {
        id: 'abc',
        title: 'dbz',
        url: 'https://localhost/dbz.gif',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
