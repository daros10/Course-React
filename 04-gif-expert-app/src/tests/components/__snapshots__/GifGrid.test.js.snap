// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifGrid Debe de mostrar items cuando se hace la llamada con el useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Rick and Morty
     
  </h3>
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        img={
          Object {
            "id": "abc",
            "title": "dbz",
            "url": "https://localhost/dbz.gif",
          }
        }
        key="abc"
      />
    </ol>
  </div>
</Fragment>
`;

exports[`Pruebas en componente GifGrid Deberia mostar el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Rick and Morty
     
  </h3>
  <div
    className="card-grid"
  >
    <p
      className="animate__animated animate__flash"
    >
      Loading...
    </p>
    <ol />
  </div>
</Fragment>
`;
